unlock-app/src/components/content/certification/CertificationForm.tsx

```diff
@@ -0,0 +1,371 @@
+import { useState } from 'react'
+import { Lock, Token } from '@unlock-protocol/types'
+import { MetadataFormData } from '~/components/interface/locks/metadata/utils'
+import { FormProvider, useForm, useWatch } from 'react-hook-form'
+import {
+  Button,
+  Disclosure,
+  Input,
+  TextBox,
+  ToggleSwitch,
+  ImageUpload,
+  Select,
+} from '@unlock-protocol/ui'
+import { useConfig } from '~/utils/withConfig'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { networkDescription } from '~/components/interface/locks/Create/elements/CreateLockForm'
+import { useQuery } from '@tanstack/react-query'
+import { useWeb3Service } from '~/utils/withWeb3Service'
+import { SelectCurrencyModal } from '~/components/interface/locks/Create/modals/SelectCurrencyModal'
+import { UNLIMITED_KEYS_DURATION } from '~/constants'
+import { CryptoIcon } from '@unlock-protocol/crypto-icon'
+import { useImageUpload } from '~/hooks/useImageUpload'
+import { BalanceWarning } from '~/components/interface/locks/Create/elements/BalanceWarning'
+// TODO replace with zod, but only once we have replaced Lock and MetadataFormData as well
+export interface NewEventForm {
+  network: number
+  lock: Omit<Lock, 'address' | 'key'>
+  currencySymbol: string
+  metadata: Partial<MetadataFormData>
+}
+
+interface FormProps {
+  onSubmit: (data: NewEventForm) => void
+}
+
+export const CertificationForm = ({ onSubmit }: FormProps) => {
+  const { networks } = useConfig()
+  const { network, account } = useAuth()
+
+  const [isFree, setIsFree] = useState(true)
+  const [unlimitedCapacity, setUnlimitedCapacity] = useState(false)
+  const [forever, setForever] = useState(false)
+  const [isCurrencyModalOpen, setCurrencyModalOpen] = useState(false)
+  const { mutateAsync: uploadImage, isLoading: isUploading } = useImageUpload()
+
+  const web3Service = useWeb3Service()
+
+  const methods = useForm<NewEventForm>({
+    mode: 'onChange',
+    shouldUnregister: false,
+    defaultValues: {
+      network,
+      lock: {
+        name: '',
+        expirationDuration: UNLIMITED_KEYS_DURATION,
```

Comment:
That is not correct. Lock managers need to be able to set a duration that is not unlimited!